interaction-lab/NRI-SVTE
# Odom branch

## Description
Fixes #17 . Fixes #19 . 

Simulated bumping with fake data using an Arrow that will point in odom direction. The Arrow is controlled by the arrow keys, and when the Arrow bumps into an obstacle the name of the obstacle collided into, sensor that is bumped into, and an array of bumper sensor statuses will be printed out.

#### Changes made

-  Used a script to feed data to move Arrow (Kuri)
     -  Can move Arrow (Kuri) forward and backward using arrow keys
     -  Can turn Arrow (Kuri) left and right using arrow keys
-  The camera follows Arrow (Kuri) as it moves - Display 2
-  Prints out bumping data
     -  When Arrow (Kuri) bumps into an object, prints out what object it has collided into
     -  Prints out which sensor is bumped into
     -  When Arrow (Kuri) bumps into an object, prints out array of values from the colliders
-  Piped these values into the visualization
     -  Only when the sensor bumps into object does the sensor change from grey (off) to a green color (on)


## How to test
- Move the Arrow using the up/down arrow keys to move forward/backward
- Turn the Arrow using the left/right arrow keys
- Have the Arrow bump into the obstacle (the cube), and see how the sensors (right, middle, left) turn green only when they bump into the obstacle
     - Name of obstacle collided into should be printed out (e.g. `Collided with Obstacle`)
     - Which sensor is bumped into is printed out (e.g. `LeftSensor position 2 has been bumped`)
     - Array of bumper sensor statuses will be printed out (e.g. `Bumper sensor status: 0 0 1`)
          - Should print out another statement when no longer touching obstacle (e.g. `Bumper sensor status: 0 0 0`)
